libs/blocks/vimeo/vimeo.js

```diff
@@ -1,25 +1,81 @@
 import { createIntersectionObserver, createTag, isInTextNode } from '../../utils/utils.js';
 
-export default function init(a) {
-  if (isInTextNode(a)) return;
-  const embedVimeo = () => {
-    const url = new URL(a.href);
-    let src = url.href;
-    if (url.hostname !== 'player.vimeo.com') {
-      const video = url.pathname.split('/')[1];
-      src = `https://player.vimeo.com/video/${video}?app_id=122963`;
-    }
-    const iframe = createTag('iframe', {
-      src,
-      style: 'border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;',
+function addPrefetch(kind, url, as) {
+  const linkElem = createTag('link', { rel: kind, href: url });
+  if (as) linkElem.as = as;
+  linkElem.crossorigin = true;
+  document.head.append(linkElem);
+}
+
+class LiteVimeo extends HTMLElement {
+  static preconnected = false;
+
+  connectedCallback() {
+    this.videoId = this.getAttribute('videoid');
+    this.language = this.getAttribute('language') || '';
+    this.setupThumbnail();
+    this.setupPlayButton();
+    this.addEventListener('pointerover', LiteVimeo.warmConnections, { once: true });
+    this.addEventListener('click', this.addIframe);
+  }
+
+  static warmConnections() {
+    if (LiteVimeo.preconnected) return;
+    LiteVimeo.preconnected = true;
+    addPrefetch('preconnect', 'https://player.vimeo.com');
+    addPrefetch('preconnect', 'https://i.vimeocdn.com');
+    addPrefetch('preconnect', 'https://f.vimeocdn.com');
+    addPrefetch('preconnect', 'https://fresnel.vimeocdn.com');
+  }
+
+  setupThumbnail() {
+    const { width, height } = this.getBoundingClientRect();
+    const roundedWidth = Math.min(Math.ceil(width / 100) * 100, 1920);
+    const roundedHeight = Math.round((roundedWidth / width) * height);
+
+    fetch(`https://vimeo.com/api/v2/video/${this.videoId}.json`)
+      .then((response) => response.json())
+      .then((data) => {
+        const thumbnailUrl = data[0].thumbnail_large?.replace(/-d_[\dx]+$/i, `-d_${roundedWidth}x${roundedHeight}`);
+        this.style.backgroundImage = `url("${thumbnailUrl}")`;
+      })
+      .catch((e) => {
+        window.lana.log(`Error fetching Vimeo thumbnail: ${e}`, { tags: 'errorType=info,module=vimeo' });
+      });
+  }
+
+  setupPlayButton() {
+    const playBtnEl = createTag('button', { type: 'button', 'aria-label': 'Play video', class: 'ltv-playbtn' });
```

Comment:
Ideally this aria-label would be translated for accessability. I think we're not even passing a label in Express though, which is also not great.

Reply:
tested it here as well btw - https://main--bacom--adobecom.hlx.live/customer-success-stories/coca-cola-case-study#watch-now

Reply:
> Ideally this aria-label would be translated for accessability. I think we're not even passing a label in Express though, which is also not great.

I've seen it a lot that this currently doesn't happen 😞

Reply:
I've added placeholders for that in my feedback commit 👍

Reply:
I don't want to contribute to that so I've added a placeholder 👍